docs/5.x/testplan.md

```diff
@@ -199,6 +195,107 @@ $ tele login -o example.gravitational.io
 - [ ] Unpin runtime from manifest and run tele build.
   - [ ] Verify latest compatible runtime from active Ops Center was selected.
 
+### Application Catalog
+
+This section covers the application catalog features. It requires an Ops Center.
+
+#### Building & Publishing
+
+- [ ] Build application images from the sample Helm charts in `assets/charts`.
+```bash
+$ tele build assets/charts/alpine-0.1.0
+$ tele build assets/charts/alpine-0.2.0
+```
+
+- [ ] Log into the Ops Center.
+```bash
+$ tele login -o ops.gravitational.io:32009
+```
+
+- [ ] Push the built application image into the Ops Center.
+```bash
+$ tele push alpine-0.1.0.tar
+```
+
+- [ ] Make sure the application is shown in the image list.
+```bash
+$ tele ls
+```
+
+- [ ] Verify the application image tarball can be downloaded from the Ops Center.
+```bash
+$ tele pull alpine:0.1.0
+```
+
+- [ ] Verify Helm chart can be searched for and downloaded from the Ops Center.
+```bash
+$ helm search alpine
+$ helm fetch ops.gravitational.io/alpine --version 0.1.0
+```
+
+- [ ] Verify one of the application's Docker images can be pulled from the Ops Center.
+```bash
+$ docker pull ops.gravitational.io:32009/alpine:3.3
+```
+
+#### Discovery
+
+- [ ] Install a cluster.
+
+- [ ] Connect the cluster to the Ops Center using [Trusted Cluster](https://gravitational.com/gravity/docs/cluster/#configuring-trusted-clusters) resource.
+
+- [ ] Verify the application can be searched for in the connected Ops Center.
+```bash
+$ gravity app search --all
+$ gravity app search alpine --all
+```
+
+#### Lifecycle
+
+- [ ] Transfer both built application images (`alpine-0.1.0.tar`, `alpine-0.2.0.tar`) onto the cluster node.
+
+- [ ] Install the application.
+```bash
+$ gravity app install alpine-0.1.0.tar
+```
+
+- [ ] Verify an application instance has been deployed.
+```bash
+$ gravity app ls
+$ kubectl get pods
+```
+
+- [ ] Install the application directly from the Ops Center.
+```bash
+$ gravity app install <opscenter-name>/alpine:0.1.0
```

Comment:
What does it mean to install the same application twice (if this is installing the same application)?

Reply:
Yeah, so we're using Helm for all application lifecycle and the way it works is when you install a chart (or "app" in our terminology) it creates a "release" to track this instance of the app. If you install the same app again, it'll create another "release", so it'll be another instance of the same app running. And so on. So you can have multiple instances of the same application running in the same cluster, upgrade/rollback them individually, etc. Here's some info on it in Helm docs: https://github.com/helm/helm/blob/master/docs/glossary.md#release.